src/relations/postgresql.py

```diff
@@ -88,6 +74,62 @@ def _on_database_relation_broken(self, event: DatabaseEvent) -> None:
         self._update_db_connections(event.relation.name, None)
         self.charm._update(event)
 
+    def update_db_relation_data_in_state(self) -> bool:
+        """Update database data from relation into peer relation databag.
+
+        Returns:
+            True if the charm should update its pebble layer, False otherwise.
+        """
+        if not self.charm.unit.is_leader():
+            return False
+
+        if not self.charm._state.is_ready():
+            return False
+
+        should_update = False
+        for rel_name in ["db", "visibility"]:
+            if self.charm.model.get_relation(rel_name) is None:
+                continue
+
+            if rel_name == "db":
+                relation_id = self.charm.db.relations[0].id
+                relation_data = self.charm.db.fetch_relation_data()[relation_id]
+            else:
```

Comment:
suggestion: Be explicit about the `rel_name`, this is potential for a bug in the future if the list in line 90 ever expands.